iot-hub/Tutorials/Routing/SimulatedDevice/Program.cs

```diff
@@ -56,11 +66,15 @@ private static async Task Main()
                 //  http://docs.microsoft.com/azure/iot-hub/tutorial-routing
 
                 Console.WriteLine("Routing Tutorial: Simulated device\n");
-                s_deviceClient = DeviceClient.Create(s_iotHubUri,
+
+                s_myDeviceId = parameters.DeviceId;
+                s_deviceKey = parameters.DeviceKey;
+
+                s_deviceClient = DeviceClient.Create(parameters.IoTHubUri,
                   new DeviceAuthenticationWithRegistrySymmetricKey(s_myDeviceId, s_deviceKey), TransportType.Mqtt);
 
                 using var cts = new CancellationTokenSource();
-                var messages = SendDeviceToCloudMessagesAsync(cts.Token);
+                Task messages = SendDeviceToCloudMessagesAsync(cts.Token);
                 Console.WriteLine("Press the Enter key to stop.");
                 Console.ReadLine();
```

Comment:
Change this out to use the console event listener, so you can await the previous task.